samples/nrf9160/fmfu_smp_svr/src/main.c

```diff
@@ -7,9 +7,41 @@
 #include <zephyr.h>
 #include <mgmt/fmfu_mgmt.h>
 #include <mgmt/fmfu_mgmt_stat.h>
+#include <nrf_modem_full_dfu.h>
 #include <modem/modem_info.h>
 #include <modem/nrf_modem_lib.h>
 #include <modem/at_cmd.h>
+#ifdef CONFIG_MCUMGR_CMD_IMG_MGMT
+#include "img_mgmt/img_mgmt.h"
+#endif
+#ifdef CONFIG_MCUMGR_CMD_OS_MGMT
+#include "os_mgmt/os_mgmt.h"
+#endif
+
+static void fmfu_callback_handler(const enum fmfu_evt_id id)
+{
+	switch (id) {
+	case FMFU_EVT_INIT:
+		/* Shutdown modem to prepare for DFU */
+		nrf_modem_lib_shutdown();
+
+		nrf_modem_lib_init(FULL_DFU_MODE);
+		struct nrf_modem_full_dfu_digest digest;
+
+		int err = nrf_modem_full_dfu_init(&digest);
+
+		if (err != 0) {
+			printk("nrf_fmfu_init failed, err: %d, errno: %d\n.",
+				err, errno);
+		}
+		break;
+	case FMFU_EVT_FINISHED:
+		break;
```

Comment:
Could we have a `printk` here stating that the update was finished? Just to have something :)

Reply:
sure